update-dois/src/main/java/org/reactome/release/updateDOIs/UpdateDOIs.java

```diff
@@ -32,26 +30,36 @@ public static void setAdaptors(MySQLAdaptor adaptorTR, MySQLAdaptor adaptorGK) {
 	}
 
 	@SuppressWarnings("unchecked")
-	public static void findAndUpdateDOIs(long authorIdTR, long authorIdGK, String pathToReport) {
+	public static void findAndUpdateDOIs(long authorIdTR, long authorIdGK, String pathToReport, boolean testMode) {
+
+		if (testMode) {
+			logger.info("Test mode is active. Outputting DOIs that can be updated");
+		}
 
 		Collection<GKInstance> doisTR;
 		Collection<GKInstance> doisGK;
 
 		// Initialize instance edits for each DB
 		String creatorFile = "org.reactome.release.updateDOIs.Main";
-		GKInstance instanceEditTR = UpdateDOIs.createInstanceEdit(UpdateDOIs.dbaTestReactome, authorIdTR, creatorFile);
-		GKInstance instanceEditGK = UpdateDOIs.createInstanceEdit(UpdateDOIs.dbaGkCentral, authorIdGK, creatorFile);
+		GKInstance instanceEditTR = null;
+		GKInstance instanceEditGK = null;
+		if (!testMode) {
+			instanceEditTR = UpdateDOIs.createInstanceEdit(UpdateDOIs.dbaTestReactome, authorIdTR, creatorFile);
+			instanceEditGK = UpdateDOIs.createInstanceEdit(UpdateDOIs.dbaGkCentral, authorIdGK, creatorFile);
+		}
 		// Gets the updated report file if it was provided for this release
-		HashMap<String, HashMap<String,String>> expectedUpdatedDOIs = UpdateDOIs.getExpectedUpdatedDOIs(pathToReport);
-		ArrayList<String> updated = new ArrayList<String>();
-		ArrayList<String> notUpdated = new ArrayList<String>();
-		int fetchHits = 0;
+		Map<String, Map<String,String>> expectedUpdatedDOIs = UpdateDOIs.getExpectedUpdatedDOIs(pathToReport);
+		if (expectedUpdatedDOIs.size() == 0) {
+			logger.warn("No DOIs listed in UpdateDOIs.report. Please add expected DOI and displayName to UpdateDOIs.report.");
+			return;
+		}
+		List<String> updated = new ArrayList<>();
+		List<String> notUpdated = new ArrayList<>();
 		try 
 		{
 			// Get all instances in Test Reactome in the Pathway table that don't have a 'doi' attribute starting with 10.3180, the Reactome DOI standard
-			 doisTR = dbaTestReactome.fetchInstanceByAttribute("Pathway", "doi", "NOT REGEXP", "^10.3180");
-			// Used during testing
-//			doisTR = dbaTestReactome.fetchInstanceByAttribute("Pathway", "DB_ID", "REGEXP", "1912408|3232118|3232142|4085377|4090294|4655427|4755510|8985947|9013694|9034015");
+			 doisTR = dbaTestReactome.fetchInstanceByAttribute(ReactomeJavaConstants.Pathway, "doi", "NOT REGEXP", "^10.3180");
```

Comment:
The 10.3180 could be extracted to a constant like "REACTOME_DOI_PREFIX" to document its meaning.

Reply:
Yea good idea